### DISCUSSION TITLE
How long does a restore take when migrating to a new server? Been over a week migrating

### DISCUSSION THREAD

**Discussion by hassan-sd** [immich-app/immich#4124] [Q&A]
Had 50k photos on a server, migrated all my other content such as plex etc to a new server.
When it comes to Immich I am following https://documentation.immich.app/docs/administration/backup-and-restore

It's been at 99% for over a week now trying to restore.  This is my second attempt, the first attempt I restarted the process after 5 days wondering if it broke.

My sql dump file is 350mb

Any ideas how I can do this migration quicker than over a week so far? 

It got to 99% the first day.

![image](https://github.com/immich-app/immich/assets/119671806/e5fb5a9f-a249-4380-a374-04af8494023d)

**Comment by jrasm91** [comment]
What is all the binary stuff before that line? Did you use gzip for the backup? Did you unzip it before importing? Can you show the full command your are using to export and import?

**Comment by hassan-sd** [comment]
> What is all the binary stuff before that line? Did you use gzip for the backup? Did you unzip it before importing? Can you show the full command your are using to export and import?

Thanks for the quick reply. While trying to get you the console output before all that binary i realised it wouldn't let me scroll up, was using windows terminal. I repeated the process in windows command prompt instead and I get invalid command. That binary code seems to just be junk or meaningless and it's to do with my windows alternative approach to Gzip / Gunzip. I have tried cygwins version of gzip and that's where the 350mb sql dump is created. I cannot use the restore command using gzip / gunzip or using 7z. Using 7z it states the archive is corrupted. 


I'm on windows so I did the following:

`docker exec -t immich_postgres pg_dumpall -c -U postgres | gzip > "dump.sql.gz"`

Then the 350 mb sql file was created, i moved to the new server, fresh install of windows. 

```
docker-compose down -v  # CAUTION! Deletes all Immich data to start from scratch.
docker-compose pull     # Update to latest version of Immich (if desired)
docker-compose create   # Create Docker containers for Immich apps without running them.
docker start immich_postgres    # Start Postgres server
sleep 10    # Wait for Postgres server to start up
.\bin\gunzip -c "dump.sql.gz" | docker exec -i immich_postgres psql -U postgres -d immich

then also tried

7z e -so "dump.sql.gz" | docker exec -i immich_postgres psql -U postgres -d immich

```

I can't get passed the restore command errors. I've cancelled it all and instead am bulk uploading through the guide here
https://immich.app/docs/features/bulk-upload#quick-start

For windows it would be nice to have a process of backup/restore that is UI based, or CLI versions that are suitable for multiple OS without the hassle that comes with gzip on windows.

**Reply by jrasm91** [reply]
You don't have to pipe it to gzip by the way. It can dump to a text file and restore from the same (without using unzip)
